lib/index.js

```diff
@@ -41,12 +42,23 @@ Server.prototype._createServer = function() {
     return;
   }
 
+  // connections
+  self._connections = {};
+
   var server = net.createServer(function(connection) { //'connection' listener
     new Connection(connection, self, {
       version: self.protocolVersion,
       authKey: self.authKey,
       protocol: self.protocol
     });
+    // assign id
+    connection.id = self.nextConnectionId++;
```

Comment:
Sorry I missed that. There's a listener in the Connection prototype for the `end` event. You can have it call

```
Server.prototype.removeConnection = function(id) {
  delete this._connections[id]
}
```